src/test/java/core/basesyntax/strategyhandler/StrategyHandlerImplTest.java

```diff
@@ -0,0 +1,58 @@
+package core.basesyntax.strategyhandler;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.storage.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class StrategyHandlerImplTest {
+    private StrategyHandlerImpl strategyHandler;
+    private Map<FruitTransaction.Operation, OperationHandler> strategyMap;
+    private List<FruitTransaction> handledTransactions;
+
+    @BeforeEach
+    void setUp() {
+        strategyHandler = new StrategyHandlerImpl();
+        handledTransactions = new ArrayList<>();
+        OperationHandler balanceHandler = transaction -> handledTransactions.add(transaction);
+        OperationHandler purchaseHandler = transaction -> handledTransactions.add(transaction);
+        OperationHandler returnHandler = transaction -> handledTransactions.add(transaction);
+        OperationHandler supplyHandler = transaction -> handledTransactions.add(transaction);
+        strategyMap = Map.of(
+                FruitTransaction.Operation.BALANCE, balanceHandler,
+                FruitTransaction.Operation.PURCHASE, purchaseHandler,
+                FruitTransaction.Operation.RETURN, returnHandler,
+                FruitTransaction.Operation.SUPPLY, supplyHandler
+        );
+    }
+
+    @Test
+    void strategyHandler_validOperation_ok() {
```

Comment:
This is why `void` methods are hard to test. I advise you to rewrite the strategy class. The method should return a handler depending on the operation. You can iterate over the handlers and call the handler method on them in the Main class or in another additional service